src/common/task_system/task_scheduler.cpp

```diff
@@ -116,6 +118,11 @@ void TaskScheduler::removeErroringTask(uint64_t scheduledTaskID) {
     lock_t lck{mtx};
     for (auto it = taskQueue.begin(); it != taskQueue.end(); ++it) {
         if (scheduledTaskID == (*it)->ID) {
+            // We need to wait for all worker threads to abort execution of the task.
+            while (!(*it)->task->isCompleted()) {
```

Comment:
I think you have the guarantee that the task has already completed when entering this function.

Reply:
yes, removed this call